Creating a user role with permissions crashes provider

**Describe the bug**
Creating a user role with any list properties defined causes the provider to crash to a type conversion issue.

**Steps to reproduce**
1. Create an `octopusdeploy_user_role` resource with either/or `granted_space_permissions`/`granted_system_permissions`
2. Run tf apply
3. Error will occur during apply

**Expected behavior**
Resource should get created

**Logs and other supporting information**
#### Resource
```terraform
resource "octopusdeploy_user_role" "name" {
  name                      = "test"
  granted_space_permissions = ["AccountCreate"]
}
```
#### Logs
```go
Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is []interface {}, not []string
2021-03-31T21:56:18.124-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.15: 
2021-03-31T21:56:18.124-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.15: goroutine 24 [running]:
2021-03-31T21:56:18.124-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.15: github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.expandUserRole(0xc0007c1500, 0xc0007eaec0)
2021-03-31T21:56:18.124-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.15:   github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/schema_user_role.go:24 +0x5b3
2021-03-31T21:56:18.124-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.15: github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.resourceUserRoleCreate(0xf79520, 0xc00007d980, 0xc0007c1500, 0xcef3c0, 0xc00035b400, 0xc0007eaec0, 0x926c0a, 0xc0007dd6a0)
2021-03-31T21:56:18.124-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.15:   github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/resource_user_role.go:25 +0x45
2021-03-31T21:56:18.124-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.15: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0001c30a0, 0xf794a0, 0xc0007bacc0, 0xc0007c1500, 0xcef3c0, 0xc00035b400, 0x0, 0x0, 0x0)
2021-03-31T21:56:18.124-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.15:   github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/schema/resource.go:330 +0x17c
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment and versions:**
- OS: OSx
- Octopus Server Version: 2020.5.8
- Terraform Version: 0.14.7
- Octopus Terraform Provider Version: 0.7.15

**Additional context**
Add any other context about the problem here.

--- a/octopusdeploy/resource_user_role_test.go
+++ b/octopusdeploy/resource_user_role_test.go
@@ -26,9 +26,33 @@ func TestAccUserRoleBasic(t *testing.T) {
 	})
 }
 
+func TestAccUserRolePermissions(t *testing.T) {
+	localName := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	name := acctest.RandStringFromCharSet(16, acctest.CharSetAlpha)
+
+	resource.Test(t, resource.TestCase{
+		CheckDestroy: testAccUserRoleCheckDestroy,
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testUserRolePermissions(localName, name),
+			},
+		},
+	})
+}
+
 func testUserRoleMinimum(localName string, name string) string {
 	return fmt.Sprintf(`resource "octopusdeploy_user_role" "%s" {
 		name = "%s"
+	}`, localName, name)
+}
+
+func testUserRolePermissions(localName string, name string) string {
+	return fmt.Sprintf(`resource "octopusdeploy_user_role" "%s" {
+		name                       = "%s"
+		granted_space_permissions  = ["AccountCreate"]
+		granted_system_permissions = ["SpaceView"]
 	}`, localName, name)
 }
 

--- a/octopusdeploy/schema_user_role.go
+++ b/octopusdeploy/schema_user_role.go
@@ -21,11 +21,11 @@ func expandUserRole(d *schema.ResourceData) *octopusdeploy.UserRole {
 	}
 
 	if v, ok := d.GetOk("granted_space_permissions"); ok {
-		userRole.GrantedSpacePermissions = v.([]string)
+		userRole.GrantedSpacePermissions = getSliceFromTerraformTypeList(v)
 	}
 
 	if v, ok := d.GetOk("granted_system_permissions"); ok {
-		userRole.GrantedSystemPermissions = v.([]string)
+		userRole.GrantedSystemPermissions = getSliceFromTerraformTypeList(v)
 	}
 
 	if v, ok := d.GetOk("name"); ok {
@@ -33,15 +33,15 @@ func expandUserRole(d *schema.ResourceData) *octopusdeploy.UserRole {
 	}
 
 	if v, ok := d.GetOk("space_permission_descriptions"); ok {
-		userRole.SpacePermissionDescriptions = v.([]string)
+		userRole.SpacePermissionDescriptions = getSliceFromTerraformTypeList(v)
 	}
 
 	if v, ok := d.GetOk("supported_restrictions"); ok {
-		userRole.SupportedRestrictions = v.([]string)
+		userRole.SupportedRestrictions = getSliceFromTerraformTypeList(v)
 	}
 
 	if v, ok := d.GetOk("system_permission_descriptions"); ok {
-		userRole.SystemPermissionDescriptions = v.([]string)
+		userRole.SystemPermissionDescriptions = getSliceFromTerraformTypeList(v)
 	}
 
 	return userRole
@@ -106,6 +106,7 @@ func getUserRoleSchema() map[string]*schema.Schema {
 		"id":   getIDSchema(),
 		"name": getNameSchema(true),
 		"space_permission_descriptions": {
+			Computed: true,
 			Elem:     &schema.Schema{Type: schema.TypeString},
 			Optional: true,
 			Type:     schema.TypeList,
@@ -116,6 +117,7 @@ func getUserRoleSchema() map[string]*schema.Schema {
 			Type:     schema.TypeList,
 		},
 		"system_permission_descriptions": {
+			Computed: true,
 			Elem:     &schema.Schema{Type: schema.TypeString},
 			Optional: true,
 			Type:     schema.TypeList,

